src/lib/components/lines-and-dots/svg/group-details-row.svelte

```diff
@@ -0,0 +1,185 @@
+    DetailsChildTimelineHeight,
+    DetailsConfig,
+    getGroupDetailsBoxHeight,
+    getStatusColor,
+    mergeEventGroupDetails,
+    staticCodeBlockHeight,
+  } from '../constants';
+
+  import Box from './box.svelte';
+  import GroupDetailsText from './group-details-text.svelte';
+  import Text from './text.svelte';
+  import TimelineGraph from './timeline-graph.svelte';
+
+  export let group: EventGroup;
+  export let canvasWidth: number;
+  export let x = 0;
+  export let y: number;
+
+  $: status = group?.finalClassification || group?.classification;
+
+  $: {
+    if (group?.pendingActivity) {
+      if (group.pendingActivity.attempt > 1) {
+        status = 'Retrying';
+      } else {
+        status = 'Pending';
+      }
+    }
+  }
+
+  const { gutter, fontSizeRatio, height } = DetailsConfig;
+  $: ({ namespace } = $page.params);
+
+  let fetchChildWorkflow;
+  let fetchChildTimeline;
+
+  const fetchChildWorkflowForGroup = () => {
+    if (group && group.category === 'child-workflow' && namespace) {
+      const completedEvent = group.eventList.find(
+        isChildWorkflowExecutionCompletedEvent,
+      );
+      const startedEvent = group.eventList.find(
+        isChildWorkflowExecutionStartedEvent,
+      );
+      const childEvent = completedEvent ?? startedEvent;
+      if (childEvent) {
+        const childWorkflowId =
+          childEvent.attributes.workflowExecution.workflowId;
+        const childRunId = childEvent.attributes.workflowExecution.runId;
+        fetchChildWorkflow = fetchWorkflow({
+          namespace,
+          workflowId: childWorkflowId,
+          runId: childRunId,
+        }).then(({ workflow }) => {
+          status = workflow?.status;
+          return workflow;
+        });
+        fetchChildTimeline = fetchAllEvents({
+          namespace,
+          workflowId: childWorkflowId,
+          runId: childRunId,
+        });
+      }
+    } else {
+      fetchChildWorkflow = undefined;
+      fetchChildTimeline = undefined;
+    }
+  };
+
+  const labelPadding = 240;
+
+  $: title = group.name;
+  $: boxHeight = getGroupDetailsBoxHeight(group);
+  $: textStartingY = height + y + fontSizeRatio;
+  $: attributes = mergeEventGroupDetails(group);
+  $: codeBlockAttributes = Object.entries(attributes).filter(
+    ([, value]) => typeof value === 'object',
+  );
+  $: textAttributes = Object.entries(attributes).filter(
+    ([, value]) => typeof value !== 'object',
+  );
+  $: childTimelineY = textStartingY + textAttributes.length * fontSizeRatio;
+  $: group, fetchChildWorkflowForGroup();
+  $: width = canvasWidth;
+</script>
+
+<g role="button" tabindex="0" class="relative cursor-pointer">
+  <Box point={[x, y]} {width} height={boxHeight} fill="#465A78" />
+  <Box point={[x, y]} {width} {height} fill="#1E293B" />
+  <Box
+    point={[x, y]}
+    width={gutter + 100}
+    {height}
+    fill={getStatusColor(status)}
+  />
+  <Text point={[x + gutter, y + 0.5 * height]} category="none">
+    {status}
```

Comment:
Looks like there's the possibility for this to be `undefined` ⬇️  (this is a Marker event)

<img width="1315" alt="Screenshot 2024-04-04 at 4 33 52 PM" src="https://github.com/temporalio/ui/assets/15069288/f75f9d98-6b67-49e2-a9de-87e80889a297">

Reply:
👍 I added a check to use group label if no status.